src/frame/mod.rs

```diff
@@ -387,22 +387,46 @@ impl Response {
 
 /// A server (slave) exception.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-#[repr(u8)]
 pub enum Exception {
-    IllegalFunction = 0x01,
-    IllegalDataAddress = 0x02,
-    IllegalDataValue = 0x03,
-    ServerDeviceFailure = 0x04,
-    Acknowledge = 0x05,
-    ServerDeviceBusy = 0x06,
-    MemoryParityError = 0x08,
-    GatewayPathUnavailable = 0x0A,
-    GatewayTargetDevice = 0x0B,
+    /// 0x01
+    IllegalFunction,
+    /// 0x02
+    IllegalDataAddress,
+    /// 0x03
+    IllegalDataValue,
+    /// 0x04
+    ServerDeviceFailure,
+    /// 0x05
+    Acknowledge,
+    /// 0x06
+    ServerDeviceBusy,
+    /// 0x08
+    MemoryParityError,
+    /// 0x0A
+    GatewayPathUnavailable,
+    /// 0x0B
+    GatewayTargetDevice,
+    /// None of the above.
+    ///
+    /// Although encoding one of the predefined values as this is possible, it is not recommended.
```

Comment:
We could then add something like the following hint:

```
Instead of instantiating this variant directly prefer to use [`Self::new()`] to prevent such ambiguities.
```